### PR TITLE
fixed issue with animation

### PR DESCRIPTION
**Pull request by OleksiiKolomiiets**

### CODE REVIEW

File: DataStructuresMemories/Models/DataSourceModel.swift [hunk 1]
```diff
@@ -76,6 +73,17 @@ class DataSourceModel: DataSource {
         return -1
     }
     
-    
-    
 }
+
+let dataStructTitlesSource = [ "Stack", "Queue", "Set", "Dequeue", "PriorityQueue", "List(Array)", "MultiSet", "Dictionary" ]
+
+let dataStructLinksSource: [String] = [
```
Comment (AlexeyTyurenkov) [inline DataStructuresMemories/Models/DataSourceModel.swift]: OMG

File: DataStructuresMemories/ViewControllers/URLPresenter.swift [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        }
+    }
+    
+    func getAtcion(by controller: DetailViewController, with type: PresentersType) -> () -> Void {
+        return {
+            let mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
+            switch type {
+            case .Cancel:
+                break
+            case .WKWebView:
```
Comment (AlexeyTyurenkov) [inline DataStructuresMemories/ViewControllers/URLPresenter.swift]: Is there a better way to do it?

File: DataStructuresMemories/ViewControllers/URLPresenter.swift [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                guard let targetController = mainStoryboard.instantiateViewController(withIdentifier: type.rawValue) as? WKWebViewController else { return }
```
Comment (AlexeyTyurenkov) [inline DataStructuresMemories/ViewControllers/URLPresenter.swift]: I guess we have to use protocols here

File: DataStructuresMemories/ViewControllers/URLPresenter.swift [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+            return someFunc()
+        case .WKWebView:
+            return someFunc()
+        case .Cancel:
+            return someFunc()
+        }
+    }
+}
+
+func someFunc() {
```
Comment (AlexeyTyurenkov) [inline DataStructuresMemories/ViewControllers/URLPresenter.swift]: WTF?

File: DataStructuresMemories/ViewControllers/WKWebViewController.swift [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+import UIKit
+import WebKit
+
+class WKWebViewController: UIViewController, WKNavigationDelegate, PresenterProtocol {
+
+   
+    var webView: WKWebView!
+    
+    override func loadView() {
```
Comment (AlexeyTyurenkov) [inline DataStructuresMemories/ViewControllers/WKWebViewController.swift]: hmmm not sure

**Review by AlexeyTyurenkov — CHANGES_REQUESTED** [review]

File: DataStructuresMemories/ViewControllers/URLPresenter.swift (hunk 4 above)
Comment (OleksiiKolomiiets) [reply DataStructuresMemories/ViewControllers/URLPresenter.swift]: I will fixed it.

File: DataStructuresMemories/Models/DataSourceModel.swift (hunk 1 above)
Comment (OleksiiKolomiiets) [reply DataStructuresMemories/Models/DataSourceModel.swift]: I will fixed it.

File: DataStructuresMemories/ViewControllers/URLPresenter.swift (hunk 3 above)
Comment (OleksiiKolomiiets) [reply DataStructuresMemories/ViewControllers/URLPresenter.swift]: Ok.

File: DataStructuresMemories/ViewControllers/URLPresenter.swift (hunk 2 above)
Comment (OleksiiKolomiiets) [reply DataStructuresMemories/ViewControllers/URLPresenter.swift]: Yes, I will fixed it)

File: DataStructuresMemories/ViewControllers/WKWebViewController.swift (hunk 5 above)
Comment (OleksiiKolomiiets) [reply DataStructuresMemories/ViewControllers/WKWebViewController.swift]: It works. I will look for a better approach.

File: DataStructuresMemories/ViewControllers/URLPresenter.swift [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    func getAtcion(by controller: DetailViewController, with type: PresentersType) -> () -> Void {
+        return {
+            let mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
+            switch type {
+            case .Cancel:
+                break
+            default:
+                guard var targetController = mainStoryboard.instantiateViewController(withIdentifier: type.rawValue) as? PresenterProtocol else { return }
+                targetController.link = controller.tappedCell?.link
+                controller.navigationController?.present(targetController as! UIViewController, animated: true)
```
Comment (AlexeyTyurenkov) [inline DataStructuresMemories/ViewControllers/URLPresenter.swift:48]: Why do you use force unwrap here?

File: DataStructuresMemories/ViewControllers/WebPresenters/SFSafaryPresenter.swift [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    var link: String?
+    
+    var isDone = false
+    
+    @IBOutlet weak var safaryView: UIWebView!
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(true)
+        if let link = link, let url = URL(string: link) {
+            let safariVC = SFSafariViewController(url: url, entersReaderIfAvailable: false)
```
Comment (AlexeyTyurenkov) [inline DataStructuresMemories/ViewControllers/WebPresenters/SFSafaryPresenter.swift:22]: why do you implement in this way?

**Review by AlexeyTyurenkov — APPROVED** [review]

**Pull request merged**